src/llscan.h

```diff
@@ -174,36 +181,71 @@ class TypeRecord {
 
 
  private:
+  friend DetailedTypeRecord;
   std::string type_name_;
   uint64_t instance_count_;
   uint64_t total_instance_size_;
   std::set<uint64_t> instances_;
 };
 
+class DetailedTypeRecord : public TypeRecord {
+ public:
+  DetailedTypeRecord(std::string& type_name, uint64_t own_descriptors_count,
+                     uint64_t indexed_properties_count)
+      : TypeRecord(type_name),
+        own_descriptors_count_(own_descriptors_count),
+        indexed_properties_count_(indexed_properties_count) {}
+  inline uint64_t GetOwnDescriptorsCount() { return own_descriptors_count_; };
+  inline uint64_t GetIndexedPropertiesCount() {
+    return indexed_properties_count_;
+  };
+
+ private:
+  std::vector<std::string> properties_;
+  uint64_t own_descriptors_count_ = 0;
+  uint64_t indexed_properties_count_ = 0;
```

Comment:
The initializers are pointless, aren't they, since they're assigned in the constructor's init list?

Reply:
Yes, they are now. They were needed in a previous version and I forgot to remove them.